content/blog/chess-knight-problem/index.md

```diff
@@ -0,0 +1,129 @@
+---
+{
+  title: "Chess Knight Problem: a quick and dirty solution in JavaScript",
+  description: "Here I present a quick and dirty solution to a common interview question where the solution is not nearly as complex as it may first appear.",
+  published: "2020-04-29T12:27:06.284Z",
+  authors: ["thodges314"],
+  tags: ["javascript", "computer science", "interview"],
+  attached: [],
+  license: "cc-by-4",
+}
+---
+
+This is a classic problem in Computer Science that is said to come up in various forms on programming interviews - given an 8 x 8 chessboard, a selected starting square, and a selected ending square, what is the minimum number of legal moves are required to get a knight piece from the starting square to the ending square? Note that this is _not_ the same as Knight's Tour, a much more involved problem where we try to find a path around the board such that the knight touches every square once without repeat.
+
+For some background - a chess knight moves in an L-shaped pattern - two up and one to the right, two to the left and one up, and so on. The knight jumps from it's starting location to it's ending location.
+
+There are many diagrams of possible knight moves on a google image search, but to avoid copyright claims, I made up my own diagram:
+![Image of possible moves for a knight centered on a chessboard.](./knight-moves-0.png)
+
+The red mark above is an arbitrary starting point, and the green marks are all of the possible places that the knight can jump from that point.
+
+# Solution Method {#solution-method}
+
+At first glance, this may look like a bizarre maze navigation algorithm with complex rules, and inspires any number of thoughts of number of possible iterations, how to decide if a move is constructive or not, etc.
+
+Happily, our solution is much simpler, such that I was able to tap one up in JavaScript in a single evening after I decided to look into this problem.
+
+If, in the diagram above, we start in the red space, all of the green spaces are places we can go in one move. Let's label out starting spot '0', and the places that we can go in one move with '1.'
+![Image of possible moves for a knight centered on a chessboard with the initial location populated by a 0 and the possible moved populated by 1s.](./knight-moves-1.png)
+
+Let's choose one of those spots and see all of the places that we can go from there. Any spot that we can reach from this spot is a place that we can reach in two moves from our starting spot. Let's label them with '2's.
+![Same image as above, but with one of the 1s highlighted, and the possible moved from there populated by 2s.  In the starting space, we see both a 0 and a 2.](./knight-moves-2.png)
```

Comment:
The overlapping 2s and 0s aren't really clear. What about something like this?

![knight-moves-2](https://user-images.githubusercontent.com/9100169/80654049-4b067880-8a30-11ea-8402-4d7a900605a3.png)

Reply:
Thanks for the work on that.  That does look clearer.  I'll swap int he image.